src/sentry/performance_issues/detectors/disable_detectors.py

```diff
@@ -0,0 +1,67 @@
+from typing import TypedDict
+
+from sentry import projectoptions
+from sentry.models.project import Project
+
+
+class DetectorDisablingConfig(TypedDict):
+    detector_project_option: str
+    disable_in_all_cases: bool
+    languages_to_disable: set[str] | None  # Should be non-None when not disable_in_all_cases
+
+
+SETTINGS_PROJECT_OPTION_KEY = "sentry:performance_issue_settings"
+
+DEFAULT_DETECTOR_DISABLING_CONFIGS: list[DetectorDisablingConfig] = [
+    {
+        "detector_project_option": "db_query_injection_detection_enabled",
+        "disable_in_all_cases": False,
+        "languages_to_disable": {
+            "php",
+            "php-laravel",
+            "php-symfony",
+            "go",
+            "go-echo",
+            "go-fasthttp",
+            "go-fiber",
+            "go-gin",
+            "go-http",
+            "go-iris",
+            "go-martini",
+            "go-negroni",
+        },
+    }
+]
+
+
+def set_default_disabled_detectors(project: Project) -> None:
+    performance_issue_settings_default = projectoptions.get_well_known_default(
+        SETTINGS_PROJECT_OPTION_KEY,
+        project=project,
+    )
+    performance_issue_settings = project.get_option(
+        SETTINGS_PROJECT_OPTION_KEY, default=performance_issue_settings_default
+    )
+
+    disabled_by_config_keys = set()
+    for disable_config in DEFAULT_DETECTOR_DISABLING_CONFIGS:
+        if disable_config["disable_in_all_cases"]:
```

Comment:
not entirely sure I understand `disable_in_all_cases` - is there a scenario where we would use this instead of just default disabling this setting in `DEFAULT_PROJECT_PERFORMANCE_DETECTION_SETTINGS`